pkg/scanner/transformer.go

```diff
@@ -12,30 +12,39 @@ type Transformer interface {
 	Transform(artifact harbor.Artifact, source aqua.ScanReport) harbor.ScanReport
 }
 
-func NewTransformer(clock clock.Clock) Transformer {
+func NewTransformer(clock ext.Clock) Transformer {
 	return &transformer{
 		clock: clock,
 	}
 }
 
 type transformer struct {
-	clock clock.Clock
+	clock ext.Clock
 }
 
 func (t *transformer) Transform(artifact harbor.Artifact, source aqua.ScanReport) harbor.ScanReport {
+	log.WithFields(log.Fields{
+		"digest":          source.Digest,
+		"image":           source.Image,
+		"summary":         source.Summary,
+		"scan_options":    source.ScanOptions,
+		"changed_results": source.ChangedResults,
+		"partial_results": source.PartialResults,
+	}).Debug("Transforming scan report")
 	var items []harbor.VulnerabilityItem
 
 	for _, resourceScan := range source.Resources {
+		var pkg string
+		switch resourceScan.Resource.Type {
+		case aqua.Library:
+			pkg = resourceScan.Resource.Path
+		case aqua.Package:
+			pkg = resourceScan.Resource.Name
+		default:
+			pkg = resourceScan.Resource.Name
+		}
+
 		for _, vln := range resourceScan.Vulnerabilities {
-			var pkg string
-			switch resourceScan.Resource.Type {
-			case aqua.Library:
-				pkg = resourceScan.Resource.Path
-			case aqua.Package:
-				pkg = resourceScan.Resource.Name
-			default:
-				continue
```

Comment:
This was the actual bug. Some resources do not have a type. They only have the format property set.